kafka-handler/src/test/org/apache/hadoop/hive/kafka/KafkaBrokerResource.java

```diff
@@ -64,7 +81,25 @@ class KafkaBrokerResource extends ExternalResource {
     brokerProps.setProperty("zookeeper.connect", zkConnect);
     brokerProps.setProperty("broker.id", "0");
     brokerProps.setProperty("log.dir", tmpLogDir.toString());
-    brokerProps.setProperty("listeners", "PLAINTEXT://" + BROKER_IP_PORT);
+    Map<String, BrokerListener> listeners = new HashMap<>();
+    listeners.put("L1", new BrokerListener(BROKER_IP_PORT, "PLAINTEXT"));
+    if (principal != null) {
+      listeners.put("L2", new BrokerListener(BROKER_SASL_PORT, "SASL_PLAINTEXT"));
+    }
+    String listenersURLs = listeners.entrySet().stream().map((e) -> e.getKey() + "://" + e.getValue().url)
+        .collect(Collectors.joining(","));
+    brokerProps.setProperty("listeners", listenersURLs);
+    String listenersProtocols = listeners.entrySet().stream().map((e) -> e.getKey() + ":" + e.getValue().protocol)
+        .collect(Collectors.joining(","));
+    brokerProps.setProperty("listener.security.protocol.map", listenersProtocols);
+    brokerProps.setProperty("inter.broker.listener.name", "L1");
+    if (principal != null) {
+      String jaasConfig = String.format("%s %s %s %s serviceName=\"%s\" keyTab=\"%s\" principal=\"%s\";",
+          "com.sun.security.auth.module.Krb5LoginModule required", "debug=true", "useKeyTab=true", "storeKey=true",
+          principal, keytab, principal + "/localhost");
+      brokerProps.setProperty("listener.name.l2.gssapi.sasl.jaas.config", jaasConfig);
```

Comment:
what is this prop used for? is a test-only thing or should be used in production too?
just asking, I cannot remember we've ever used it

Reply:
This is for setting up the Kafka broker with a valid SASL config for the respective listener (i.e., L2). If want the Kafka broker to user Kerberos then we have to provide a valid config. It is a test-only thing cause only in tests we need to configure and start Kafka brokers.

See https://kafka.apache.org/documentation/#security_jaas_broker for more.